### PR TITLE
fix subnet acl with same net allow

### LINKED ISSUE #3960: [BUG] In some scenarios, allowing the same subnet cannot take effect
### Kube-OVN Version

v1.12.12

### Kubernetes Version

v.1.21.5

### Operation-system/Kernel Version

CentOS Linux 7 (Core) 3.10.0-1160.92.1.el7.x86_64

### Description

ovn v22.12.3
ovs 3.1.5

subnet1 10.0.2.0/24
vm1 192.168.1.4
vm2 192.168.1.5

subnet ACL configure as:
from-lport  1900 (ip4.src == 10.0.2.0/24 && ip4.dst == 10.0.2.0/24) allow-related
  to-lport  1000 (ip4.src == 10.0.2.4 && ip4.dst == 10.0.2.5) drop
  to-lport  1000 (ip4.src == 10.0.2.5 && ip4.dst == 10.0.2.4) drop

vm1 ping vm2 is unreachable

subnet ACL configure as:
from-lport  1900 (ip4.src == 10.0.2.0/24 && ip4.dst == 10.0.2.0/24) allow-related
from-lport  1000 (ip4.src == 10.0.2.4 && ip4.dst == 10.0.2.5) drop
  to-lport  1000 (ip4.src == 10.0.2.5 && ip4.dst == 10.0.2.4) drop

vm1 ping vm2 is reachable

### Steps To Reproduce

as description

### Current Behavior

vm1 ping vm2 is unreachable

### Expected Behavior

vm1 ping vm2 is reachable

### PR DESCRIPTION
**Pull request by lynn901**
# Pull Request

- [] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Bug fixes

## Which issue(s) this PR fixes

Fixes #3960

### CODE REVIEW

File: pkg/ovs/ovn-nb-acl.go [hunk 1]
```diff
@@ -440,13 +440,19 @@ func (c *OVNNbClient) UpdateLogicalSwitchACL(lsName, cidrBlock string, subnetAcl
 				NewACLMatch(ipSuffix+".dst", "==", cidr, ""),
 			)
 
-			sameSubnetACL, err := c.newACL(lsName, ovnnb.ACLDirectionToLport, util.AllowEWTrafficPriority, sameSubnetMatch.String(), ovnnb.ACLActionAllowRelated, options)
-			if err != nil {
-				klog.Error(err)
-				return fmt.Errorf("new same subnet ingress acl for logical switch %s: %v", lsName, err)
+			ingressSameSubnetACL, ingressErr := c.newACL(lsName, ovnnb.ACLDirectionToLport, util.AllowEWTrafficPriority, sameSubnetMatch.String(), ovnnb.ACLActionAllow, options)
```
Comment (bobz965) [inline pkg/ovs/ovn-nb-acl.go:443]:
```suggestion
			ingressSameSubnetACL, err := c.newACL(lsName, ovnnb.ACLDirectionToLport, util.AllowEWTrafficPriority, sameSubnetMatch.String(), ovnnb.ACLActionAllow, options)
```

File: pkg/ovs/ovn-nb-acl.go [hunk 2]
```diff
@@ -440,13 +440,19 @@ func (c *OVNNbClient) UpdateLogicalSwitchACL(lsName, cidrBlock string, subnetAcl
...
+			if ingressErr != nil {
+				klog.Error(ingressErr)
+				return fmt.Errorf("new same subnet ingress acl for logical switch %s: %v", lsName, ingressErr)
 			}
+			acls = append(acls, ingressSameSubnetACL)
 
-			acls = append(acls, sameSubnetACL)
+			egressSameSubnetACL, EgressErr := c.newACL(lsName, ovnnb.ACLDirectionFromLport, util.AllowEWTrafficPriority, sameSubnetMatch.String(), ovnnb.ACLActionAllow, options)
```
Comment (bobz965) [inline pkg/ovs/ovn-nb-acl.go:450]:
```suggestion
			egressSameSubnetACL, err := c.newACL(lsName, ovnnb.ACLDirectionFromLport, util.AllowEWTrafficPriority, sameSubnetMatch.String(), ovnnb.ACLActionAllow, options)
```

File: pkg/ovs/ovn-nb-acl.go (hunk 1 above)
Comment (lynn901) [reply pkg/ovs/ovn-nb-acl.go:443]: err := func will not able to use

File: pkg/ovs/ovn-nb-acl.go (hunk 2 above)
Comment (lynn901) [reply pkg/ovs/ovn-nb-acl.go:450]: err := func can't use twice

**Pull request merged**